src/legacy/core_plugins/kibana/public/discover/np_ready/angular/context/query_parameters/actions.test.ts

```diff
@@ -0,0 +1,161 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { getQueryParameterActions } from './actions';
+import {
+  FilterManager,
+  IndexPatternsContract,
+} from '../../../../../../../../../plugins/data/public';
+import { coreMock } from '../../../../../../../../../core/public/mocks';
+const setupMock = coreMock.createSetup();
+
+let state: {
+  queryParameters: {
+    defaultStepSize: number;
+    indexPatternId: string;
+    predecessorCount: number;
+    successorCount: number;
+  };
+};
+let filterManager: FilterManager;
+let indexPatterns: IndexPatternsContract;
+let filterManagerSpy: jest.SpyInstance;
+
+beforeEach(() => {
+  filterManager = new FilterManager(setupMock.uiSettings);
+  filterManagerSpy = jest.spyOn(filterManager, 'addFilters');
+
+  state = {
+    queryParameters: {
+      defaultStepSize: 3,
+      indexPatternId: 'INDEX_PATTERN_ID',
+      predecessorCount: 10,
+      successorCount: 10,
+    },
+  };
+});
+
+describe('context query_parameter actions', function() {
+  describe('action addFilter', () => {
+    it('should pass the given arguments to the filterManager', () => {
+      const { addFilter } = getQueryParameterActions(filterManager);
+
+      addFilter(state)('FIELD_NAME', 'FIELD_VALUE', 'FILTER_OPERATION');
+
+      // get the generated filter
+      const generatedFilter = filterManagerSpy.mock.calls[0][0][0];
+      const queryKeys = Object.keys(generatedFilter.query.match_phrase);
+      expect(filterManagerSpy.mock.calls.length).toBe(1);
+      expect(queryKeys[0]).toBe('FIELD_NAME');
+      expect(generatedFilter.query.match_phrase[queryKeys[0]]).toBe('FIELD_VALUE');
+    });
+
+    it('should pass the index pattern id to the filterManager', () => {
+      const { addFilter } = getQueryParameterActions(filterManager);
+      addFilter(state)('FIELD_NAME', 'FIELD_VALUE', 'FILTER_OPERATION');
+      const generatedFilter = filterManagerSpy.mock.calls[0][0][0];
+      expect(generatedFilter.meta.index).toBe('INDEX_PATTERN_ID');
+    });
+  });
+  describe('action setPredecessorCount', () => {
+    it('should set the predecessorCount to the given value', () => {
+      const { setPredecessorCount } = getQueryParameterActions(filterManager);
+      setPredecessorCount(state)(20);
+      expect(state.queryParameters.predecessorCount).toBe(20);
+    });
+
+    it('should limit the predecessorCount to 0 as a lower bound', () => {
+      const { setPredecessorCount } = getQueryParameterActions(indexPatterns);
```

Comment:
Shouldn't `getQueryParameterActions` take the filter manager as first argument?

Reply:
Good catch, yes it should, and now it does. Wasn't used in the setPredecessorCount method, so test didn't fail